gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -103,8 +107,10 @@
   private final AtomicReference<ForkState> forkState;
 
   public Fork(TaskContext taskContext, TaskState taskState, Object schema, int branches, int index,
-      CountDownLatch countDownLatch)
-      throws Exception {
+      CountDownLatch countDownLatch) throws Exception {
+
+    super(taskState, Fork.class.getClass(), createMetricsTags(taskState, index));
+    Instrumented.setMetricContextName(taskState, this.getMetricContext().getName());
```

Comment:
You need to register the metric context before doing this (JobMetrics and TaskMetrics do it by default, but not Instrumented). You can do `GobblinMetricsRegistry.getInstance().putIfAbsent(context.getName(), context)`.